pkg/cmd/server/kubernetes/node/options/options.go

```diff
@@ -0,0 +1,211 @@
+// an error
+func Build(options configapi.NodeConfig) (*kubeletoptions.KubeletServer, *componentconfig.KubeProxyConfiguration, error) {
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = options.ImageConfig.Format
+	imageTemplate.Latest = options.ImageConfig.Latest
+
+	var path string
+	var fileCheckInterval int64
+	if options.PodManifestConfig != nil {
+		path = options.PodManifestConfig.Path
+		fileCheckInterval = options.PodManifestConfig.FileCheckIntervalSeconds
+	}
+
+	kubeAddressStr, kubePortStr, err := net.SplitHostPort(options.ServingInfo.BindAddress)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot parse node address: %v", err)
+	}
+	kubePort, err := strconv.Atoi(kubePortStr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot parse node port: %v", err)
+	}
+
+	// Defaults are tested in TestKubeletDefaults
+	server := kubeletoptions.NewKubeletServer()
+	// Adjust defaults
+	server.RequireKubeConfig = true
+	server.KubeConfig.Default(options.MasterKubeConfig)
+	server.PodManifestPath = path
+	server.RootDirectory = options.VolumeDirectory
+	server.NodeIP = options.NodeIP
+	server.HostnameOverride = options.NodeName
+	server.AllowPrivileged = true
+	server.RegisterNode = true
+	server.Address = kubeAddressStr
+	server.Port = int32(kubePort)
+	server.ReadOnlyPort = 0        // no read only access
+	server.CAdvisorPort = 0        // no unsecured cadvisor access
+	server.HealthzPort = 0         // no unsecured healthz access
+	server.HealthzBindAddress = "" // no unsecured healthz access
+	server.ClusterDNS = []string{options.DNSIP}
+	server.ClusterDomain = options.DNSDomain
+	server.NetworkPluginName = options.NetworkConfig.NetworkPluginName
+	server.HostNetworkSources = []string{kubelettypes.ApiserverSource, kubelettypes.FileSource}
+	server.HostPIDSources = []string{kubelettypes.ApiserverSource, kubelettypes.FileSource}
+	server.HostIPCSources = []string{kubelettypes.ApiserverSource, kubelettypes.FileSource}
+	server.HTTPCheckFrequency = metav1.Duration{Duration: time.Duration(0)} // no remote HTTP pod creation access
+	server.FileCheckFrequency = metav1.Duration{Duration: time.Duration(fileCheckInterval) * time.Second}
+	server.KubeletFlags.ContainerRuntimeOptions.PodSandboxImage = imageTemplate.ExpandOrDie("pod")
+	server.LowDiskSpaceThresholdMB = 256 // this the previous default
+	server.CPUCFSQuota = true            // enable cpu cfs quota enforcement by default
+	server.MaxPods = 250
+	server.PodsPerCore = 10
+	server.CgroupDriver = "systemd"
+	server.DockerExecHandlerName = string(options.DockerConfig.ExecHandlerName)
+	server.RemoteRuntimeEndpoint = options.DockerConfig.DockerShimSocket
+	server.RemoteImageEndpoint = options.DockerConfig.DockerShimSocket
+	server.DockershimRootDirectory = options.DockerConfig.DockershimRootDirectory
+
+	if network.IsOpenShiftNetworkPlugin(server.NetworkPluginName) {
+		// set defaults for openshift-sdn
+		server.HairpinMode = componentconfig.HairpinNone
```

Comment:
This can be dropped; it gets set in the `IsOpenShiftNetworkPlugin()` block below as well. (It looks like it was redundant in the old code too, we just hadn't noticed.)

Reply:
Ok